gradle-examples/gradle-kts-example-publish/build.gradle.kts

```diff
@@ -0,0 +1,132 @@
+import org.jfrog.gradle.plugin.artifactory.task.ArtifactoryTask
+
+buildscript {
+    repositories {
+        mavenLocal()
+        jcenter()
+    }
+    dependencies {
+        classpath("org.jfrog.buildinfo", "build-info-extractor-gradle", "4.+")
+    }
+    configurations.classpath {
+        resolutionStrategy {
+            cacheDynamicVersionsFor(0, "seconds")
+            cacheChangingModulesFor(0, "seconds")
+        }
+    }
+}
+
+plugins {
+    java
+    `maven-publish`
+}
+
+fun javaProjects() = subprojects.filter {
+    File(it.projectDir, "src").isDirectory
+}
+
+val currentVersion: String by project
+
+allprojects {
+    apply(plugin = "com.jfrog.artifactory")
+
+    group = "org.jfrog.test.gradle.publish"
+    version = currentVersion
+    status = "Integration"
+
+    repositories {
+        maven("http://127.0.0.1:8081/artifactory/jcenter")
+    }
+}
+
+tasks.named<ArtifactoryTask>("artifactoryPublish") {
+    skip = true
+}
+
+project("services") {
+    tasks.named<ArtifactoryTask>("artifactoryPublish") {
+        skip = true
+    }
+}
+
+configure(javaProjects()) {
+    apply(plugin = "java")
+    apply(plugin = "maven-publish")
+
+    dependencies {
+        testImplementation("junit:junit:4.7")
+    }
+
+    configure<PublishingExtension> {
+        publications {
+            register<MavenPublication>("mavenJava") {
+                from(components.getByName("java"))
+                artifact(file("$rootDir/gradle.properties"))
+            }
+        }
+    }
+}
+
+project("api") {
+    apply(plugin = "ivy-publish")
+
+    configure<PublishingExtension> {
+        publications {
+            register<IvyPublication>("ivyJava") {
+                from(components.getByName("java"))
+
+                artifact(file("$rootDir/settings.gradle.kts")) {
+                    name = "gradle-settings"
+                    extension = "txt"
+                    type = "text"
+                }
+                // The config below will add a extra attribute to the ivy.xml
+                // See http://ant.apache.org/ivy/history/latest-milestone/concept.html#extra
+                descriptor.withXml {
+                    val info = asNode().get("info") as groovy.util.NodeList
+                    val first = info.first() as groovy.util.Node
+                    first.attributes()["e:architecture"] = "amd64"
+                }
+            }
+        }
+    }
+
+    tasks.named<ArtifactoryTask>("artifactoryPublish") {
+        publications(publishing.publications["ivyJava"])
+    }
+}
+
+configure<org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention> {
+    clientConfig.isIncludeEnvVars = true
+
+    setContextUrl("http://127.0.0.1:8081/artifactory")
+    publish {
```

Comment:
I don't understand, how it works for you? Look's like Kotlin DSL have no such contructions.

Reply:
@clayly happy that this is something you're interested in! The Gradle Kotlin DSL example won't work until my other PR to enable such Kotlin DSL support is released:
https://github.com/jfrog/build-info/pull/441

Reply:
In the meantime, you _can_ still work around this by using the approach load out in this issue:

https://github.com/jfrog/build-info/issues/184

It's very ugly though.

Reply:
Ok, i got it. I wonder what is happened that there is still no Kotlin DSL support. Keep it going!